plugins/package-managers/composer/src/main/kotlin/Composer.kt

```diff
@@ -139,7 +137,11 @@ class Composer(
         }
 
         val lockfile = stashDirectories(workingDir.resolve("vendor")).use { _ ->
-            ensureLockfile(workingDir).let {
+            val lockfileProvider = LockfileProvider(definitionFile)
```

Comment:
I might be wrong, but to my understanding ORT's analyzers normally do not do this kind of clean-up.
If true, what's the reason for doing this here?

Reply:
I agree that this kind of cleanup is probably done inconsistently / not done at all for package managers currently. But I believe we should come up with a policy to start doing it consistently. Besides the annoyance of modified files when analyzing locally, leaving modified files behind also creates problems in CI pipelines that use `git describe`-based mechanisms to calculate the project version after analysis, as such approaches would eventually add a `-dirty` suffix to the version number.

Reply:
> These could be also easily addressed with dropping file changes via `git`.

Our premise has always been that the ORT analyzer can be used without changes to your build, and that includes making fixups like resetting modified files with Git in a CI pipeline in my view.

Note that approving this PR does not mean that you agree to do the same changes for other package managers. As such I'd appreciate if the discussion could be limited to the code in this PR, and to pursue the general discussion elsewhere.

Reply:
In my opinion we should always try to not modify the project files with the analyzer, and we have to, clean up afterward. For example, in NPM we could simply delete the `node_modules` directory, but instead we move it away so that it can be restored after the analysis.